lib/Conversion/TorchToLinalg/Utils.cpp

```diff
@@ -67,6 +68,45 @@ Value torch_to_linalg::getZeroPaddedTensor(
   return getPaddedTensor(op, b, input, paddingInts, paddingInts, c0);
 }
 
+// Helper function that adds dynamic padding to a tensor, ignoring the first two
+// dimensions
+Value torch_to_linalg::getDynamicZeroPaddedTensor(
+    Operation *op, OpBuilder &b, Value &input,
+    SmallVectorImpl<Value> &padding) {
+  assert(input.getType().isa<RankedTensorType>() &&
+         "input must be RankedTensorType");
+  Location loc = op->getLoc();
+  Type elementType = input.getType().cast<RankedTensorType>().getElementType();
+  auto inputRank = input.getType().cast<RankedTensorType>().getRank();
+  Value c0 = b.create<arith::ConstantOp>(loc, b.getI64IntegerAttr(0));
+  Value c1 = b.create<arith::ConstantOp>(loc, b.getIndexAttr(1));
+  Value c2 = b.create<arith::ConstantOp>(loc, b.getI64IntegerAttr(2));
+  SmallVector<Value> inputDims = getTensorSizes(b, loc, input);
+  SmallVector<Value> outputDims{inputDims[0], inputDims[1]};
+  SmallVector<Value> strides(inputRank, c1);
+  SmallVector<Value> paddingIncludingUnchanged{c0, c0};
+  paddingIncludingUnchanged.append(padding);
+  for (auto pad = paddingIncludingUnchanged.begin();
+       pad < paddingIncludingUnchanged.end(); pad++)
+    *pad = castIntToIndex(b, loc, *pad);
+
+  for (auto i = 0; i < inputRank - 2; i++) {
```

Comment:
Please refer to the implementation of `getPaddedTensor`  https://github.com/llvm/torch-mlir/blob/52c9987115b39c2f6673315a21f6cdff0d1f51ca/lib/Conversion/TorchToLinalg/Utils.cpp#L42-L52. We should be able to use `tensor::createPadScalarOp` here as well.